### PR TITLE
--skip-covered: Proof of concept

### PR DESCRIPTION
**Pull request by matthiask**
`interrogate` is great! I'm using it to check whether all unit tests have docstrings (not my idea) and it has been very helpful.

I'm running it together with black, flake8 and isort (https://github.com/matthiask/workbench/blob/master/tox.ini). Currently there does not seem to be a way to make `interrogate` only show files where coverage isn't 100%. Without a `-v` it only shows the summary, with one `-v` it also shows all files with `100%` coverage. 

This draft pull request adds a `--skip-covered` option (name and help text copied from `coverage`). Are you interested in this feature? If yes, do you see a better way to implement this? I don't want to drop a big pull request with tests and everything without starting a discussion first.

### CODE REVIEW

**Comment by rytilahti** [comment]
:+1: this would be very useful to allow CI (or a pre-commit hook) to show why it is failing.

File: src/interrogate/cli.py [hunk 1]
```diff
@@ -42,6 +42,13 @@
     show_default=True,
     help="Fail when coverage % is less than a given amount.",
 )
+@click.option(
+    "--skip-covered",
```
Comment (econchick) [inline src/interrogate/cli.py:46]: perhaps add a short flag, `-S` or `-C` aren't used yet.

**Comment by matthiask** [comment]
Some edge cases still seem to exist.

image

I don't really have an idea what's going on here right now.

**Comment by econchick** [comment]
@matthiask no problem! I will take a deeper look in the next day or so to get this sorted out.

**Comment by matthiask** [comment]
Oh right, those are the functions using nested functions. The nested functions themselves do not have docstrings, but according to `--whitelist-regex` those functions should be ignored anyway. 

image

Passing `--ignore-nested-functions` does not help either.

**Comment by econchick** [comment]
@matthiask you're right, this is a little more complicated than initially thought. It might take me some time (i.e. after work and such) to dig into the best approach. How about this: are you willing to make a new PR (leave this PR for me to work off of) with code/tests/docs for just the summary view? so we an at least get that in.

**Comment by matthiask** [comment]
@econchick I still want to work on this and I'll happily open a new pull request. I have been swamped with work-related stuff this week so I didn't even manage to reply but I hope to resume work on this soon (within days, not weeks or months 😅)

**Comment by matthiask** [comment]
`--omit-covered-files` is available now.

**Pull request closed without merging**